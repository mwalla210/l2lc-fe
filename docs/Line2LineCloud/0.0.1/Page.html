<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Page - Documentation</title>

    <script src="scripts/prettify/prettify.js"></script>
    <script src="scripts/prettify/lang-css.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc.css">
</head>
<body>

<input type="checkbox" id="nav-trigger" class="nav-trigger" />
<label for="nav-trigger" class="navicon-button x">
  <div class="navicon"></div>
</label>

<label for="nav-trigger" class="overlay"></label>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="CustomerModel.html">CustomerModel</a><ul class='methods'><li data-type='method'><a href="CustomerModel.html#edit">edit</a></li><li data-type='method'><a href="CustomerModel.html#getformattedBillAddress">getformattedBillAddress</a></li><li data-type='method'><a href="CustomerModel.html#getformattedShipAddress">getformattedShipAddress</a></li><li data-type='method'><a href="CustomerModel.html#getProjects">getProjects</a></li><li data-type='method'><a href="CustomerModel.html#setBillIsSame">setBillIsSame</a></li></ul></li><li><a href="EmployeeModel.html">EmployeeModel</a><ul class='methods'><li data-type='method'><a href="EmployeeModel.html#activate">activate</a></li><li data-type='method'><a href="EmployeeModel.html#changeName">changeName</a></li><li data-type='method'><a href="EmployeeModel.html#deactivate">deactivate</a></li></ul></li><li><a href="FormModel.html">FormModel</a><ul class='methods'><li data-type='method'><a href="FormModel.html#buttonDisabled">buttonDisabled</a></li><li data-type='method'><a href="FormModel.html#modifyFieldValue">modifyFieldValue</a></li><li data-type='method'><a href="FormModel.html#primaryButtonWrapper">primaryButtonWrapper</a></li></ul></li><li><a href="Page.html">Page</a><ul class='methods'><li data-type='method'><a href="Page.html#accountManagementMenuItem">accountManagementMenuItem</a></li><li data-type='method'><a href="Page.html#costCenterTimeMenuItem">costCenterTimeMenuItem</a></li><li data-type='method'><a href="Page.html#createNewProjMenuItem">createNewProjMenuItem</a></li><li data-type='method'><a href="Page.html#customerEditPage">customerEditPage</a></li><li data-type='method'><a href="Page.html#customerInfoMenuItem">customerInfoMenuItem</a></li><li data-type='method'><a href="Page.html#customerSummaryPage">customerSummaryPage</a></li><li data-type='method'><a href="Page.html#employeeInformationMenuItem">employeeInformationMenuItem</a></li><li data-type='method'><a href="Page.html#emplProductivityMenuItem">emplProductivityMenuItem</a></li><li data-type='method'><a href="Page.html#jobTypeProductivityMenuItem">jobTypeProductivityMenuItem</a></li><li data-type='method'><a href="Page.html#logOut">logOut</a></li><li data-type='method'><a href="Page.html#newCustomerPage">newCustomerPage</a></li><li data-type='method'><a href="Page.html#newEmployeePage">newEmployeePage</a></li><li data-type='method'><a href="Page.html#projectEditPage">projectEditPage</a></li><li data-type='method'><a href="Page.html#projectsMenuItem">projectsMenuItem</a></li><li data-type='method'><a href="Page.html#projectSummaryPage">projectSummaryPage</a></li><li data-type='method'><a href="Page.html#projectTimeEntryMenuItem">projectTimeEntryMenuItem</a></li><li data-type='method'><a href="Page.html#selectCustomerPage">selectCustomerPage</a></li><li data-type='method'><a href="Page.html#setFormModel">setFormModel</a></li><li data-type='method'><a href="Page.html#setTableModel">setTableModel</a></li><li data-type='method'><a href="Page.html#workstationTrackingMenuItem">workstationTrackingMenuItem</a></li></ul></li><li><a href="ProjectModel.html">ProjectModel</a><ul class='methods'><li data-type='method'><a href="ProjectModel.html#addRework">addRework</a></li><li data-type='method'><a href="ProjectModel.html#addTask">addTask</a></li><li data-type='method'><a href="ProjectModel.html#addTimeEntry">addTimeEntry</a></li><li data-type='method'><a href="ProjectModel.html#changeCustomer">changeCustomer</a></li><li data-type='method'><a href="ProjectModel.html#changeHold">changeHold</a></li><li data-type='method'><a href="ProjectModel.html#customerID">customerID</a></li><li data-type='method'><a href="ProjectModel.html#delete">delete</a></li><li data-type='method'><a href="ProjectModel.html#edit">edit</a></li><li data-type='method'><a href="ProjectModel.html#finish">finish</a></li><li data-type='method'><a href="ProjectModel.html#getDefaultTaskList">getDefaultTaskList</a></li><li data-type='method'><a href="ProjectModel.html#isOpen">isOpen</a></li><li data-type='method'><a href="ProjectModel.html#moveTask">moveTask</a></li><li data-type='method'><a href="ProjectModel.html#removeTask">removeTask</a></li><li data-type='method'><a href="ProjectModel.html#status">status</a></li><li data-type='method'><a href="ProjectModel.html#timeSpent">timeSpent</a></li><li data-type='method'><a href="ProjectModel.html#toggleTask">toggleTask</a></li></ul></li><li><a href="TableModel.html">TableModel</a><ul class='methods'><li data-type='method'><a href="TableModel.html#closeModal">closeModal</a></li><li data-type='method'><a href="TableModel.html#confirmAndClose">confirmAndClose</a></li><li data-type='method'><a href="TableModel.html#dataFetch">dataFetch</a></li><li data-type='method'><a href="TableModel.html#loadingOff">loadingOff</a></li><li data-type='method'><a href="TableModel.html#loadingOn">loadingOn</a></li><li data-type='method'><a href="TableModel.html#openModal">openModal</a></li></ul></li><li><a href="TaskModel.html">TaskModel</a><ul class='methods'><li data-type='method'><a href="TaskModel.html#toggle">toggle</a></li></ul></li><li><a href="UserModel.html">UserModel</a><ul class='methods'><li data-type='method'><a href="UserModel.html#changePassword">changePassword</a></li><li data-type='method'><a href="UserModel.html#login">login</a></li><li data-type='method'><a href="UserModel.html#logout">logout</a></li><li data-type='method'><a href="UserModel.html#toggleAdmin">toggleAdmin</a></li></ul></li><li><a href="Website.html">Website</a><ul class='methods'><li data-type='method'><a href="Website.html#createCustomer">createCustomer</a></li><li data-type='method'><a href="Website.html#createEmployee">createEmployee</a></li><li data-type='method'><a href="Website.html#createProject">createProject</a></li><li data-type='method'><a href="Website.html#generateBarcode">generateBarcode</a></li><li data-type='method'><a href="Website.html#getAccounts">getAccounts</a></li><li data-type='method'><a href="Website.html#getProject">getProject</a></li><li data-type='method'><a href="Website.html#setCustomer">setCustomer</a></li><li data-type='method'><a href="Website.html#setEmployee">setEmployee</a></li><li data-type='method'><a href="Website.html#setProject">setProject</a></li></ul></li></ul>
</nav>

<div id="main">
    
    <h1 class="page-title">Page</h1>
    

    




<section>

<header>
    
        <h2>
        Page
        </h2>
        
            <div class="class-description"><p>Main MobX store for page</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    <h4 class="name" id="Page"><span class="type-signature"></span>new Page<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line12">line 12</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>



    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        
        <th>Attributes</th>
        

        
        <th>Default</th>
        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>loggedin</code></td>
            

            <td class="type">
            
                
<span class="param-type">Boolean</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>false</code>
                
                </td>
            

            <td class="description last"><p>Indicates whether currently logged in</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>title</code></td>
            

            <td class="type">
            
                
<span class="param-type">String</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>'Default Title'</code>
                
                </td>
            

            <td class="description last"><p>Page title</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>content</code></td>
            

            <td class="type">
            
                
<span class="param-type">Object</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Page inner content</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>tableModel</code></td>
            

            <td class="type">
            
                
<span class="param-type"><a href="TableModel.html">TableModel</a></span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Page table model, if any</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>table</code></td>
            

            <td class="type">
            
                
<span class="param-type">Component</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Page-specific table component to render, if any</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>formModel</code></td>
            

            <td class="type">
            
                
<span class="param-type"><a href="FormModel.html">FormModel</a></span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Page form model, if any</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>formEdit</code></td>
            

            <td class="type">
            
                
<span class="param-type">Boolean</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>false</code>
                
                </td>
            

            <td class="description last"><p>Indicator for form editing</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>form</code></td>
            

            <td class="type">
            
                
<span class="param-type">Component</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Page-specific table component to render, if any</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>buttons</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>[]</code>
                
                </td>
            

            <td class="description last"><p>Page button set</p></td>
        </tr>

    
    </tbody>
</table>





























    
    </div>

    

    

    

     

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    <h4 class="name" id="accountManagementMenuItem">@action <span class="type-signature"></span>accountManagementMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line344">line 344</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Account Management page.</p>
</div>
























        
            

    

    <h4 class="name" id="costCenterTimeMenuItem">@action <span class="type-signature"></span>costCenterTimeMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line296">line 296</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Cost Center Time page.</p>
</div>
























        
            

    

    <h4 class="name" id="createNewProjMenuItem">@action <span class="type-signature"></span>createNewProjMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line63">line 63</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Create New Project page</p>
</div>
























        
            

    

    <h4 class="name" id="customerEditPage">@action <span class="type-signature"></span>customerEditPage<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line234">line 234</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Customer Edit page</p>
</div>
























        
            

    

    <h4 class="name" id="customerInfoMenuItem">@action <span class="type-signature"></span>customerInfoMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line194">line 194</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Customer Information page.</p>
</div>
























        
            

    

    <h4 class="name" id="customerSummaryPage">@action <span class="type-signature"></span>customerSummaryPage<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line209">line 209</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Customer Summary page</p>
</div>
























        
            

    

    <h4 class="name" id="employeeInformationMenuItem">@action <span class="type-signature"></span>employeeInformationMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line313">line 313</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Employee Information page.</p>
</div>
























        
            

    

    <h4 class="name" id="emplProductivityMenuItem">@action <span class="type-signature"></span>emplProductivityMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line251">line 251</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Employee Productivity page.</p>
</div>
























        
            

    

    <h4 class="name" id="jobTypeProductivityMenuItem">@action <span class="type-signature"></span>jobTypeProductivityMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line281">line 281</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Job Type Productivity page.</p>
</div>
























        
            

    

    <h4 class="name" id="logOut">@action <span class="type-signature"></span>logOut<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line365">line 365</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Log out method</p>
</div>
























        
            

    

    <h4 class="name" id="newCustomerPage">@action <span class="type-signature"></span>newCustomerPage<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line94">line 94</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for New Customer page.</p>
</div>
























        
            

    

    <h4 class="name" id="newEmployeePage">@action <span class="type-signature"></span>newEmployeePage<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line328">line 328</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for New Employee page.</p>
</div>
























        
            

    

    <h4 class="name" id="projectEditPage">@action <span class="type-signature"></span>projectEditPage<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line147">line 147</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Project Edit page</p>
</div>
























        
            

    

    <h4 class="name" id="projectsMenuItem">@action <span class="type-signature"></span>projectsMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line163">line 163</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Projects page</p>
</div>
























        
            

    

    <h4 class="name" id="projectSummaryPage">@action <span class="type-signature"></span>projectSummaryPage<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line110">line 110</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Project Summary page</p>
</div>
























        
            

    

    <h4 class="name" id="projectTimeEntryMenuItem">@action <span class="type-signature"></span>projectTimeEntryMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line178">line 178</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Project Time Entry page.</p>
</div>
























        
            

    

    <h4 class="name" id="selectCustomerPage">@action <span class="type-signature"></span>selectCustomerPage<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line79">line 79</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Select Customer page.</p>
</div>
























        
            

    

    <h4 class="name" id="setFormModel">@action <span class="type-signature"></span>setFormModel<span class="signature">(formModel)</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line51">line 51</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Sets form model</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>formModel</code></td>
            

            <td class="type">
            
                
<span class="param-type"><a href="FormModel.html">FormModel</a></span>


            
            </td>

            

            

            <td class="description last"><p>FormModel to use for page</p></td>
        </tr>

    
    </tbody>
</table>















        
            

    

    <h4 class="name" id="setTableModel">@action <span class="type-signature"></span>setTableModel<span class="signature">(tableModel)</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line42">line 42</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Sets table model</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>tableModel</code></td>
            

            <td class="type">
            
                
<span class="param-type"><a href="TableModel.html">TableModel</a></span>


            
            </td>

            

            

            <td class="description last"><p>TableModel to use for page</p></td>
        </tr>

    
    </tbody>
</table>















        
            

    

    <h4 class="name" id="workstationTrackingMenuItem">@action <span class="type-signature"></span>workstationTrackingMenuItem<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_page.js.html">store/page.js</a>, <a href="store_page.js.html#line266">line 266</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
</dl>





<div class="description">
    <p>Updates title, form, table, content, and buttons for Workstation Tracking page.</p>
</div>
























        
    

    

    
</article>

</section>




</div>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sat Feb 17 2018 02:44:47 GMT-0500 (EST) using the <a href="https://github.com/clenemt/docdash">docdash</a> theme.
</footer>

<script>prettyPrint();</script>
<script src="scripts/linenumber.js"></script>
</body>
</html>