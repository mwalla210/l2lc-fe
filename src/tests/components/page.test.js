import React from 'react'
import renderer from 'react-test-renderer'
import Page from '../../components/page'
jest.mock('../../components/login')
jest.mock('../../components/navbar')

const defaultOptions = {
  page: {
    content:null,
    title: 'title',
  },
  website: {
    currentUser:null,
  },
}

const alternateOptions = {
  page: {
    content:'Form',
    title: 'title',
  },
  website: {
    currentUser:'currentUser',
  },
}

describe('Page', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Page {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders with snapshot, alternate arguements', () => {
    let options = Object.assign({}, alternateOptions)
    const component = renderer.create(
      <Page {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
