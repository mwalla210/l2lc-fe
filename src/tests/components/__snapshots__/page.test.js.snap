// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Renders with snapshot 1`] = `<div />`;

exports[`Page Renders with snapshot, alternate arguements 1`] = `
<div>
  <div>
    <h1>
      title
    </h1>
    <Form />
  </div>
</div>
`;
